contrib/hadoop-store-builder/src/java/voldemort/store/readonly/mr/HadoopStoreBuilder.java

```diff
@@ -289,14 +237,37 @@ public void build() {
                 conf.setReducerClass(AvroStoreBuilderReducer.class);
             }
 
-            logger.info("Number of chunks: " + numChunks + ", number of reducers: " + numReducers
-                        + ", save keys: " + saveKeys + ", reducerPerBucket: " + reducerPerBucket
-                        + ", buildPrimaryReplicasOnly: " + buildPrimaryReplicasOnly);
             logger.info("Building store...");
-            RunningJob job = JobClient.runJob(conf);
 
-            // Once the job has completed log the counter
-            Counters counters = job.getCounters();
+            // The snipped below copied and adapted from: JobClient.runJob(conf);
+            // We have more control in the error handling this way.
+
+            JobClient jc = new JobClient(conf);
+            RunningJob runningJob = jc.submitJob(conf);
+            Counters counters = runningJob.getCounters();
+
+            try {
+                if (!jc.monitorAndPrintJob(conf, runningJob)) {
+                    // For some datasets, the number of chunks that we calculated is inadequate.
+                    // Here, we try to identify if this is the case.
+                    long mapOutputBytes = counters.getCounter(Task.Counter.MAP_OUTPUT_BYTES);
+                    int numChunks = conf.getInt(AbstractStoreBuilderConfigurable.NUM_CHUNKS, 1);
+                    long averageNumberOfBytesPerChunk = mapOutputBytes / numChunks / cluster.getNumberOfPartitions();
+                    if (averageNumberOfBytesPerChunk > (HadoopStoreWriter.DEFAULT_CHUNK_SIZE)) {
+                        float chunkSizeBloat = averageNumberOfBytesPerChunk / (float) HadoopStoreWriter.DEFAULT_CHUNK_SIZE;
```

Comment:
I imagine you meant `Float` rather than `Bloat`?

Also, aren't you calculating a number of chunks here, rather than a chunk size?

I suggest renaming to `properNumChunksFloat` or something like that.